Bumping pysam to 0.13 *core*, without customization

dannon opened the pull request:
Htslib and pysam now support flexible index names/paths, so we can finally kill https://github.com/dannon/pysam/tree/flexindex_84.  This depends on https://github.com/galaxyproject/starforge/pull/143 being fixed and merged (which I am working on).

dannon committed:
Bumping pysam to 0.12 *core*, without customization

dannon commented:
(rebased to fix conflicts)

nsoranzo commented on lib/galaxy/datatypes/converters/vcf_to_vcf_bgzip.py:
```diff
@@ -8,8 +8,7 @@
 """
 import optparse
-
-from pysam import ctabix
```
I think you should keep the blank line, `optparse` is part of the standard library, while `pysam` is not.

dannon replied:
Yep, agreed, I'll tweak it in a sec.

dannon committed:
Separate non-stdlib pysam from other imports.

dannon commented:
Wheel still under construction, there was a library inclusion fail there.  Will retest when it's re-published.

natefoo committed:
Explicitly pin pysam to 0.12.0.1

natefoo committed:
Explicitly include PyPI as an --extra-index-url to pip so we can install
from PyPI even when a package exists (but the correct version or
architecture does not exist) on wheels.galaxyproject.org. Previously,
PyPI was only checked after wheels.galaxyproject.org because pypiserver
performs an automatic redirect to PyPI for any packages that do not
exist in its index.

nsoranzo commented on lib/galaxy/datatypes/converters/interval_to_tabix_converter.py:
```diff
@@ -26,13 +26,13 @@ def main():
-                           keep_original=True, index_filename=out_fname)
+        pysam.tabix_index(filename=index_fname, seq_col=(options.chrom_col - 1),
+                          start_col=(options.start_col - 1), end_col=(options.end_col - 1),
+                          keep_original=True, index_filename=out_fname)
```
It looks like thay `tabix_index` in https://github.com/pysam-developers/pysam/blob/master/pysam/libctabix.pyx still doesn't support `keep_original` and `index_filename` parameters.

mvdbeek replied:
That's unfortunate, because it looks like it's almost there: https://github.com/pysam-developers/pysam/blob/master/pysam/libchtslib.pxd#L1355.

I'll try to PR this.

mvdbeek replied:
I see there are also some other enhancements in @dannon's branch, like detecting gzip compression instead of relying on the file ending, which would would probably need to include as well. Have we ever tried contributing this back ?

mvdbeek replied:
let's see how this goes: https://github.com/pysam-developers/pysam/pull/537

dannon replied:
Good catch, I guess I shouldn't have assumed both index methods had been updated :/

If you want, I can PR the gzip detection stuff.

Back when I wrote this, it would have required changes to htslib *and* pysam, and I never got around to doing it.

mvdbeek replied:
> If you want, I can PR the gzip detection stuff.

That would be great. There's also the `keep_original` setting, which I think we need as well.

natefoo commented:
macOS and x32 Linux wheels are on wheels.galaxyproject.org via galaxyproject/starforge#161.

Although it sounds like we'll have to update again...

dannon commented:
We'll need to wait for https://github.com/pysam-developers/pysam/pull/537, (merge and then release), and then we can proceed here.  Changing milestone to 18.01.

mvdbeek commented:
Alright, we got a new release https://pypi.python.org/pypi/pysam/0.13!

dannon commented:
@mvdbeek Nice!  Last time the wheel building was a bit of a pain, so that may take a bit of effort, but I'll go ahead and update this branch tomorrow and test against 0.13

nsoranzo commented:
Can we use the wheels on PyPI? https://pypi.python.org/pypi/pysam

dannon commented:
@nsoranzo Yes, that's what we were going to do when we were attempting to go to 0.12.01.  We still have to build and host the osx wheel, though.

dannon committed:
Merge remote-tracking branch 'upstream/dev' into pysam012

dannon committed:
All the way to 0.13 for pysam.

dannon commented:
Non-pypi'd wheels PR open at https://github.com/galaxyproject/starforge/pull/178, though I was able to run this branch on my osx box (automatically fetched source from pypi and built fine).  I *think* the selenium test failure is unrelated (passes fine on my local machine).

mvdbeek reviewed (approved):
Awesome, thanks a lot @dannon !

natefoo commented:
:trumpet: <sup>:trumpet:</sup> <sub>:trumpet:</sub> :trumpet:
